pkg/policy/ebpf.go

```diff
@@ -379,11 +378,11 @@ func (ps *Policies) createNewEventsMapVersion(
 	// store pointer to the new inner map version
 	ps.bpfInnerMaps[innerMapName] = newInnerMap
 
-	for id, ecfg := range eventsState {
+	for id, ecfg := range ps.evtsStates.states {
```

Comment:
ditto about deadlocks.

Reply:
You should probably still use the `GetAll()` function, no?

Reply:
`GetAll()` returns a new map (copy).

> IMO it will be cleaner to use the API to avoid dependencies on internal implementation.

WDYT for those specific cases we rely on a private API like `getAll()`, so its syntax automatically flag us that we're on a locked/controlled state?

Reply:
> GetAll() returns a new map (copy).

This is why I asked about performance.
If you really want to reduce performance, you could maybe create a type which envelope a map with read-only methods. This is a bit of an overkill maybe, idk.
I think that using the `getAll()` is better then direct access anyways.

Reply:
Example for iterator design pattern in Go - https://refactoring.guru/design-patterns/iterator/go/example#example-0

Reply:
That pattern is interesting, let's keep it in mind.